src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -410,63 +404,62 @@ def sample_prior(
-                                    keys = [val[0] for val in gen_kw_node.items()]
-                                    df = df.set_index(df.columns[0])
-                                    vals = df.reindex(keys).values.flatten()
-                                else:
-                                    vals = df.values.flatten()
-                            else:
-                                vals = rng.standard_normal(len(gen_kw_node))
-                            s = io.BytesIO()
-                            # The first element is time_t (64 bit integer), but
-                            # it is not used so we write 0 instead - for
-                            # padding purposes.
-                            s.write(struct.pack("Qi", 0, int(ErtImplType.GEN_KW)))
-                            s.write(vals.tobytes())
-
-                            _clib.enkf_fs.write_parameter(
-                                storage,
-                                config_node.getKey(),
-                                realization_nr,
-                                s.getvalue(),
-                            )
-
-                    else:
-                        _clib.enkf_state.state_initialize(
-                            enkf_node,
-                            storage,
-                            realization_nr,
-                        )
 
+        for parameter in parameters:
+            config_node = self.ensembleConfig().getNode(parameter)
+            enkf_node = EnkfNode(config_node)
+            if config_node.getUseForwardInit():
+                continue
+            impl_type = enkf_node.getImplType()
+            if impl_type == ErtImplType.GEN_KW:
+                gen_kw_node = enkf_node.asGenKw()
+                if config_node.get_init_file_fmt():
+                    df_values = pd.DataFrame()
+                    for iens in active_realizations:
+                        df = pd.read_csv(
+                            config_node.get_init_file_fmt() % iens,
+                            delim_whitespace=True,
+                            header=None,
+                        )
+                        # This means we have a key: value mapping in the
+                        # file otherwise it is just a list of values
+                        if df.shape[1] == 2:
+                            # We need to sort the user input keys by the
+                            # internal order of sub-parameters:
+                            keys = [val[0] for val in gen_kw_node.items()]
+                            df = df.set_index(df.columns[0])
+                            values = df.reindex(keys).values.flatten()
+                        else:
+                            values = df.values.flatten()
+                        df_values[f"{iens}"] = values
+                    parameter_values = df_values.values
+                else:
+                    parameter_values = []
+                    for key, _ in gen_kw_node.items():
```

Comment:
```suggestion
                    for key in gen_kw_node:
```

Reply:
This is sadly not possible, `gen_kw_node` does not behave as a dict.

Reply:
I realized that it might be the case after writing the comment :sweat_smile: